app/helpers/template_helpers.py

```diff
@@ -14,215 +14,299 @@
+
+        if self.context_options.footer_links:
+            for footer_link in self.context_options.footer_links:
+                footer_link.url = self._base_url + footer_link.url
+            context["rows"] = [{"itemsList": self.context_options.footer_links}]
+
+        if self.context_options.footer_legal_links:
+            for footer_legal_link in self.context_options.footer_legal_links:
+                url = footer_legal_link.url
+                footer_legal_link.url = self._base_url + url
+            context["legal"] = [{"itemsList": self.context_options.footer_legal_links}]
+
+        if (
+            self.context_options.powered_by_logo
+            or self.context_options.powered_by_logo_alt
+        ):
+            context["poweredBy"] = {
+                "logo": self.context_options.powered_by_logo,
+                "alt": self.context_options.powered_by_logo_alt,
+            }
+
+        return context
+
+    @cached_property
+    def _footer_warning(self) -> Union[str, None]:
+        if request.blueprint == "post_submission":
+            return lazy_gettext(
+                "Make sure you <a href='{sign_out_url}'>leave this page</a> or close your browser if using a shared device"
+            ).format(sign_out_url=self._sign_out_url)
+
+    @staticmethod
+    def _map_theme(theme: str) -> str:
+        """Maps a survey schema theme to a design system theme
+
+        :param theme: A schema defined theme
+        :returns: A design system theme
+        """
+        if theme and theme not in ["census", "census-nisra"]:
+            return "main"
+        return "census"
+
+
+@dataclass
+class ContextOptionsCymraeg(ContextOptions):
+    footer_links: Iterable[Link] = field(
+        default_factory=lambda: [
+            Link(
+                lazy_gettext("Help"),
+                "help/sut-i-ateb-y-cwestiynau/help-y-cwestiynau-ar-lein/",
+            ),
+            Link(lazy_gettext("Contact us"), "cysylltu-a-ni/"),
+            Link(lazy_gettext("Languages"), "help/ieithoedd-a-hygyrchedd/ieithoedd/"),
+            Link(
+                lazy_gettext("BSL and audio videos"),
+                "help/ieithoedd-a-hygyrchedd/hygyrchedd/fideos-hygyrch-gyda-bsl/",
+            ),
+        ]
+    )
+    footer_legal_links: Iterable[Link] = field(
+        default_factory=lambda: [
+            Link(lazy_gettext("Cookies"), "cwcis/"),
+            Link(lazy_gettext("Accessibility Statement"), "datganiad-hygyrchedd/"),
+            Link(
+                lazy_gettext("Privacy and data protection"),
+                "preifatrwydd-a-diogelu-data/",
+            ),
+            Link(lazy_gettext("Terms and conditions"), "telerau-ac-amodau/"),
+        ]
+    )
 
 
-def _map_theme(theme: str) -> str:
-    """Maps a survey schema theme to a design system theme
+@dataclass
+class CensusContextOptions(ContextOptions):
+    title_logo: str = "census-logo-en"
+    title_logo_alt: str = lazy_gettext("Census 2021")
```

Comment:
I can't see where the links are defined for Census?

Reply:
Added in 3142831e7